packages/validator/src/capability.js

```diff
@@ -8,15 +8,37 @@ import {
   Failure,
 } from './error.js'
 import { invoke, delegate } from '@ucanto/core'
+import * as Schema from './schema.js'
+
+/**
+ * @template {API.Ability} A
+ * @template {API.URI} R
+ * @template {API.Caveats} C
+ * @typedef {{
+ * can: A
+ * with: API.Reader<R, API.Resource, API.Failure>
+ * nb?: Schema.MapRepresentation<C, unknown>
+ * derives?: (claim: {can:A, with: R, nb: C}, proof:{can:A, with:R, nb:C}) => API.Result<true, API.Failure>
+ * }} Descriptor
+ */
 
 /**
  * @template {API.Ability} A
  * @template {API.URI} R
  * @template {API.Caveats} [C={}]
- * @param {API.Descriptor<A, R, C>} descriptor
+ * @param {Descriptor<A, R, C>} descriptor
+ 
  * @returns {API.TheCapabilityParser<API.CapabilityMatch<A, R, C>>}
  */
-export const capability = descriptor => new Capability(descriptor)
+export const capability = ({
+  derives = defaultDerives,
+  nb = defaultNBSchema,
+  ...etc
+}) => new Capability({ derives, nb, ...etc })
+
+const defaultNBSchema =
+  /** @type {Schema.MapRepresentation<any>} */
```

Comment:
```suggestion
  /** @type {Schema.MapRepresentation<unknown>} */
```

not sure if this makes sense but just an idea..

Reply:
No this is not going to work I'm afraid because it's used as default parameter for generic object schema and anything but `any` would fail to meet that requirement.

Ideally TS would respect default type parameter and allow matching it with default parameter in the function, but sadly it doesn't work so this is the only workaround I'm aware of